metrics/exp/exp.go

```diff
@@ -134,6 +134,18 @@ func (exp *exp) publishTimer(name string, metric metrics.Timer) {
 	exp.getFloat(name + ".mean-rate").Set(t.RateMean())
 }
 
+func (exp *exp) publishResettingTimer(name string, metric metrics.ResettingTimer) {
+	t := metric.Snapshot()
```

Comment:
Note that we are calling `metric.Snapshot` which collects the current values and resets the counters. This means that we cannot use multiple reporters on a given resetting timer, as a call to `Snapshot` resets the timer.